### PR TITLE
chore: Cleanup ticket ratings

### PR DESCRIPTION
**Pull request by Mugenor**
- Drop unused code
- Anchor ephemeral message reply to a query instead of ticket message. Slack works the same way in both cases, but the former is the intended way.
- Fix calculation for isEscalated property
- Moved inputs for rating buttons from actionId string, to a `value` input
- Make transactional only the rating saving part

### CODE REVIEW

File: api/service/src/main/java/com/coreeng/supportbot/rating/handler/RatingActionHandler.java [hunk 1]
```diff
@@ -54,101 +52,26 @@ public void apply(BlockActionRequest req, ActionContext context) throws IOExcept
...
-                
-            } else {
-                log.warn("Invalid rating action format after prefix removal: {}", remainder);
-            }
-        } catch (Exception e) {
-            log.error("Error handling rating submission", e);
-            throw e;
+        RatingButtonInput input = mapper.parseButtonInput(action.getValue());
+        Rating rating = service.save(input.ticketId(), input.rating());
+        if (rating != null) {
```
[the remark this answers is not in the document]
Comment (Mugenor) [reply api/service/src/main/java/com/coreeng/supportbot/rating/handler/RatingActionHandler.java]: We do it in case it's submitted as expected.
`rating == null` in case ticket is not found, or a rating was already submitted
both of the cases are corner cases and should never happen unless something weird happened

File: api/service/src/test/java/com/coreeng/supportbot/rating/RatingServiceSaveTest.java [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+        assertThat(saved.tags()).containsExactly("ui");
+        assertThat(saved.isEscalated()).isTrue();
+
+        Rating persisted = ratingRepository.findById(saved.id());
+        assertThat(persisted).isNotNull();
+        assertThat(persisted.isEscalated()).isTrue();
+
+        assertThat(ticketRepository.isTicketRated(created.id())).isTrue();
+    }
+}
```
[the remark this answers is not in the document]
Comment (Mugenor) [reply api/service/src/test/java/com/coreeng/supportbot/rating/RatingServiceSaveTest.java]: You will need to add a mock for posting ephemeral messages, other than that, it should be pretty straightforward

File: api/service/src/main/java/com/coreeng/supportbot/escalation/EscalationInMemoryRepository.java [hunk 3]
```diff
@@ -91,6 +91,12 @@ public boolean existsByThreadTs(MessageTs threadTs) {
         return escalationsByThreadTs.containsKey(threadTs);
     }
 
+    @Override
```
[the remark this answers is not in the document]
Comment (Mugenor) [reply api/service/src/main/java/com/coreeng/supportbot/escalation/EscalationInMemoryRepository.java]: These in memory repositories are only used in tests, and the logic there is trivial, so I  would leave it for simplicity
If we were to have tests for jdbc repositories, it would be nice to have them shared between repositories, but we don't have them and no plans for it at the moment

**Review by SennaSemakula — APPROVED** [review]
Looks good!
